src/storage/patch/RepresentationPatchHandler.ts

```diff
@@ -37,10 +46,65 @@ export class RepresentationPatchHandler extends PatchHandler {
-
+    // Converter of RDF text to N3 store
+    const converter = new RdfToQuadConverter();
+    let clonedRepresentation;
+    if (representation) {
+      clonedRepresentation = await cloneRepresentation(representation);
+    }
     // Patch it
     const patched = await this.patcher.handleSafe({ patch, identifier, representation });
 
+    // Solid, §5.3: "Servers MUST NOT allow HTTP POST, PUT and PATCH to update a container’s resource metadata
+    // statements; if the server receives such a request, it MUST respond with a 409 status code.
+    // https://solid.github.io/specification/protocol#contained-resource-metadata-statements
+    if (isContainerIdentifier(identifier)) {
+      throw new ConflictHttpError('Not allowed to execute PATCH request on containers.');
+    }
+
+    if (identifier.path.endsWith('.meta')) {
+      // Convert RDF representation to N3
+      const patchedClone = await cloneRepresentation(patched);
+      const patchedRepresentationQuads = await converter.handleSafe({
+        identifier,
+        representation: patchedClone,
+        preferences: { type: { [INTERNAL_QUADS]: 1 }},
+      });
+      const patchedStore = await readableToQuads(patchedRepresentationQuads.data);
+
+      if (clonedRepresentation) {
+        const representationQuads = await converter.handleSafe({
+          identifier,
+          representation: clonedRepresentation,
+          preferences: { type: { [INTERNAL_QUADS]: 1 }},
+        });
+        const representationStore = await readableToQuads(representationQuads.data);
+        // Test whether pim has been altered
+        const pimStorageIds = representationStore.getSubjects(RDF.type, PIM.Storage, null)
+          .map((subject: Term): string => subject.id);
+        const pimStoragePatchedIds = patchedStore.getSubjects(RDF.type, PIM.Storage, null)
+          .map((subject: Term): string => subject.id);
+
+        let pimStorageNotChanged = true;
+        pimStorageNotChanged = pimStorageNotChanged && pimStorageIds.length === pimStoragePatchedIds.length;
+        for (const pimStorageId of pimStorageIds) {
+          pimStorageNotChanged = pimStoragePatchedIds.includes(pimStorageId) ? pimStorageNotChanged : false;
+        }
+        if (!pimStorageNotChanged) {
+          throw new BadRequestHttpError('Not allowed to change metadata that contains PIM storage.');
+        }
+      } else if (patchedStore.countQuads(null, RDF.type, PIM.Storage, null) !== 0) {
+        // In patched, now pim or ldp:contains may be present and that is not allowed
+        throw new BadRequestHttpError('Not allowed to change metadata that contains PIM Storage.');
+      }
+      // Test whether ldp:contains was added in the metadata
+      // (remember that ldp:contains is normally build based on resources present in the store,
+      // thus it has to be checked when the source existed already or not)
+      const ldpContainsPresent = patchedStore.countQuads(null, LDP.contains, null, null) === 0;
```

Comment:
This class is not aware of how the `ldp:contains` triples are generated. Just that they are not allowed to be changed. Wouldn't this make it so that this code errors every time we try to patch a container metadata resource that has at least 1 `ldp:contains` triple?